src/watchdog.rs

```diff
@@ -0,0 +1,98 @@
+use crate::hal::watchdog::{Watchdog, WatchdogEnable};
+
+use crate::stm32::{DBGMCU, IWDG};
+
+const LSI_KHZ: u32 = 40;
+const MAX_PR: u8 = 8;
+const MAX_RL: u16 = 0x1000;
+const KR_ACCESS: u16 = 0x5555;
+const KR_RELOAD: u16 = 0xAAAA;
+const KR_START: u16 = 0xCCCC;
+
+struct IndependentWatchDog {
+    iwdg: IWDG,
+}
+
+impl IndependentWatchDog {
+    pub fn new(iwdg: IWDG) -> Self {
+        IndependentWatchdog { iwdg }
```

Comment:
```suggestion
        IndependentWatchDog { iwdg }
```

Hard to miss, took me some time :sweat_smile: There are other instances too.

Reply:
Well in this instance particular, you could even use `Self`